File: Beecrowd/2-AD-HOC/1250-KiloMan-2/1250.py
repeat = int(input())
for _ in range(repeat):
    ans = 0
    size = int(input())
    disparos = list(map(int, input().split()))
    jumps = input()

    for i in range(size):
        if (jumps[i] == 'S' and disparos[i] <= 2) or (jumps[i] == 'J' and disparos[i] > 2):
            ans += 1

    print(ans)

File: Beecrowd/2-AD-HOC/1250-KiloMan-2/1250.cs
using System;

class Program {
    static void Main() {
        int repeat = int.Parse(Console.ReadLine());
        for (int r = 0; r < repeat; r++) {
            int ans = 0;
            int size = int.Parse(Console.ReadLine());
            int[] disparos = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
            string jumps = Console.ReadLine();

            for (int i = 0; i < size; i++) {
                if ((jumps[i] == 'S' && disparos[i] <= 2) || 
                    (jumps[i] == 'J' && disparos[i] > 2)) {
                    ans++;
                }
            }
            Console.WriteLine(ans);
        }
    }
}
